core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletDataMap.java

```diff
@@ -491,6 +520,23 @@ public boolean isScanRequired(FilterResolverIntf filterExp) {
     return blocklets;
   }
 
+  @Override public List<Blocklet> prune(FilterResolverIntf filterExp, List<String> partitions) {
+    List<String> storedPartitions = getPartitions();
```

Comment:
provide comments for the logic inside this function

Reply:
ok, added